Modules/itertoolsmodule.c

```diff
@@ -2666,6 +2666,14 @@ combinations_traverse(combinationsobject *co, visitproc visit, void *arg)
     return 0;
 }
 
+/* macro that maps a combination index value to its pool element */
+#define SET_COMBINATION_VALUE(i)                \
+    elem = PyTuple_GET_ITEM(pool, indices[i]);  \
+    Py_INCREF(elem);                            \
+    oldelem = PyTuple_GET_ITEM(result, i);      \
+    PyTuple_SET_ITEM(result, i, elem);          \
+    Py_DECREF(oldelem);
+
```

Comment:
Please move this macro down to where it is used (for example, just above `combinations_next()`), and remember to `#undef` it after use.

Reply:
Also, it is generally better to write such macros using the C macro idiom `#define MACRO(...) do { /* body */ while (0)`. See PEP-7 for more details.